### ISSUE TITLE
a user should be able to unsubscribe from a card

### ISSUE THREAD

**Issue by isaacvetter** [cds-hooks/docs#41]
In discussion with @brynrhodes and @kpshek regarding aligning Clinical Reasoning and CDS Hooks, one of the topics of discussion was around the ability of a user to "unsubscribe" from a card.

It would make a lot of clinicians happy if they were able to unsubscribe from a card. This information would also be valuable to CDS services.

It would make sense to enhance the analytics-endpoint to support this additional information.

Overall, the analytics-endpoint that's currently defined in the CDS Hooks specification has not yet seen significant adoption, although it's clearly a useful feature to CDS services.

**Comment by bdoolittle** [comment]
I agree that that clinicians will undoubtedly want to have some control in the filtering of their CDS cards and that CDS services will find the user feedback valuable. 

If the hooks spec supports an unsubscribe, should it also support a way for clinicians to subscribe to content (or content subscription may be performed outside the scope of the CDS-hooks workflow)?

Should a clinician be able to unsubscribe from any card? This gives the user more control, but may also prevent important feedback from being delivered at the point of care. Does this introduce a patient safety risk?

**Comment by isaacvetter** [comment]
Hey @bdoolittle ,

Great questions. 

>support a way for clinicians to subscribe to content 

I think that an EHR could technically already support end-user chosen dynamic selection of cds services. If the EHR or healthsystem simply pre-populated discovery endpoints, it would be theoretically possible for the EHR to present all services to the end-user for "subscriptions".

>Should a clinician be able to unsubscribe from any card? This gives the user more control, but may also prevent important feedback from being delivered at the point of care. Does this introduce a patient safety risk?

The above question also came up a few times in the CDS and CQI working group meetings at the HL7 WGM in Madrid this week. Yes, indeed, supporting a straight, simple end-user unsubscribe could be dangerous or at least very likely to break healthsystem policy. In reality, an unsubscribe feature should be carefully supported by the EHR per service, based upon the purpose of the service and the clinical and business policies. Since the desire to support unsubscribe ought not to be decided by the service, it should be up to the EHR to display the option to unsubscribe from a card based upon careful implementation and/or configuration. 

The EHR could implement the unsubscribe itself, and simply stop calling the service based upon user preferences; however, the feedback implicit in an unsubscribe is valuable to the service.

All the above means that the spec itself needn't worry about placing guardrails around unsubscribe. The question is: how could we structure the analytics endpoint to support this and other desired feedback to the service.

Isaac

**Comment by robs16** [comment]
I am unclear what is meant by unsubscribing to a 'card'. Are you referring to the card instance - essentially the same as ignoring a particular issue, or the type of card, such as a card with a specific summary (or some other similar filter criteria)?
